site/docs/expressions/extended_expression.md

```diff
@@ -1,19 +1,25 @@
 # Extended expression
 
-Extended expression is provided for expression level protocol instead of plan rels. It mainly targets for expression only evaluation, such as those computed in Filter/Project/Aggregation rels. Different from original expression defined in substrait protocol, it requires more information to completely describe the computation context, including input data schema, referred function signatures and output schema. 
+Extended expression is a top-level message, like a plan, but for expressions. They can hold expressions to be used in filter, projection, and aggregation operations that aren't part of a full plan. Because they are a top-level message, they include more context than expressions that exist _within_ a plan. Some of this is additional computational context, such as the input data schema and the output schema. And some are message metadata, such as the Substrait version and the URIs of referenced extensions.
 
-Besides, as it will be used seperately with plan rel representation, it need include basic fields like Version.
+For details, see the definition in [extended_expression.proto](https://github.com/substrait-io/substrait/blob/main/proto/substrait/extended_expression.proto).
+
+## Referred expression
+
+Extended expressions can contain one or multiple expressions. A single expression is used for filters, while multiple expressions are used for projections and aggregations. Each of these expressions is represented by an `ExpressionReference` message, which can either contain an [Expression](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto) or [AggregateFunction](https://github.com/substrait-io/substrait/blob/7f272f13f22cd5f5842baea42bcf7961e6251881/proto/substrait/algebra.proto#L1170) message. In the future, other types might be allowed to handle different use cases.
```

Comment:
> A single expression is used for filters, while multiple expressions are used for projections and aggregations.

This seems overly prescriptive.  For example, I feel like this implies you would only use this for one of those three things.  However, someone might want to use this for expression evaluation that is completely unrelated to execution plans or engines.

Maybe something along the lines of "if multiple expressions are specified then there is no implicit relationship between the different top-level expressions.  Each expression should be considered independent and operates on the same top-level schema."

Reply:
Perhaps I can offer that as example usage, but not present as only valid use?

Reply:
But I will also add that note about the expressions not depending on each other.